### PR TITLE
support assertionTemplate operations on array root

### LINKED ISSUE #434: Unable to insert into assetion template array
**Bug** 

The `.insertBefore`/`.insertAfter` methods should not throw a `TypeError: Cannot read property 'children' of undefined` error when trying to insert at the top level of an assertion template. This is caused because the found element doesn't have a parent.

Package Version: 6.0.0-alpha.8

**Code**
```typescript
const template = assertionTemplate(() => [
    <div assertion-key="first" />
    <div />
]);
template.insertAfter('~first', () => (<div />));
```

**Expected behavior:**
The element(s) should be inserted at the top level of the template

**Actual behavior:**
Error

> TypeError: Cannot read property 'children' of undefined

**Suggested Fix**
The insert operation should replace `parent.children` references with a variable that could either be the parent children or the top level array.

https://github.com/dojo/framework/blob/a41f807af9204a9d0739d1fa8e6d92d3d9e97254/src/testing/assertionTemplate.ts#L139-L159

### PR DESCRIPTION
**Pull request by aciccarello**
**Type:** bug

Allows insert, remove, and replace operations on nodes in a root array of an assertion template.

* [x] There is a related issue
* [x] All code has been formatted with [`prettier`](https://prettier.io/) as per the [readme code style guidelines](./../#code-style)
* [x] Unit or Functional tests are included in the PR

**Description:**

Resolves #434

### CODE REVIEW

File: tests/testing/unit/assertionTemplate.tsx [hunk 1]
```diff
@@ -142,6 +161,24 @@ describe('assertionTemplate', () => {
 		h.expect(childAssertion);
 	});
 
+	it('can remove a node in the root', () => {
+		const h = harness(() => w(MultiRootWidget, { last: false }));
+		const insertionAssertion = baseMultiRootAssertion.remove('~last');
+		h.expect(insertionAssertion);
+	});
+
+	it('can replacee a node', () => {
```
Comment (matt-gadd) [inline tests/testing/unit/assertionTemplate.tsx]: is this `remove` too? 😄

Comment (aciccarello) [reply tests/testing/unit/assertionTemplate.tsx]: You don't think I meant `replacee`? 😆

**Comment by aciccarello** [comment]
@matt-gadd I've renamed that test

File: tests/testing/unit/assertionTemplate.tsx [hunk 2]
```diff
@@ -142,6 +161,24 @@ describe('assertionTemplate', () => {
 		h.expect(childAssertion);
 	});
 
+	it('can remove a node in the root', () => {
+		const h = harness(() => w(MultiRootWidget, { last: false }));
+		const insertionAssertion = baseMultiRootAssertion.remove('~last');
+		h.expect(insertionAssertion);
+	});
+
+	it('can remove a node', () => {
```
Comment (agubler) [inline tests/testing/unit/assertionTemplate.tsx]: @aciccarello this test seems to be duplicated and is causing intern to get upset... the exact same test is two tests above, could have been accidentally left there when merging a conflicts perhaps?

Comment (aciccarello) [reply tests/testing/unit/assertionTemplate.tsx]: @agubler As I look at that more, I think that was meant to be a replace node test. Sorry. 😞

**Pull request merged**